localstack-core/localstack/utils/docker_utils.py

```diff
@@ -147,23 +146,22 @@ def container_ports_can_be_bound(
         port_mappings.add(port.port, port.port, protocol=port.protocol)
     try:
         result = DOCKER_CLIENT.run_container(
-            _get_ports_check_docker_image(),
-            entrypoint="sh",
-            command=["-c", "echo test123"],
+            "bash",
+            command=["echo", "test123"],
             ports=port_mappings,
             remove=True,
         )
     except Exception as e:
-        if "port is already allocated" not in str(e) and "address already in use" not in str(e):
+        if "address already in use" not in str(e):
```

Comment:
The exact error message probably changed at some point, so if a port is already in use, it does not say `port is already allocated`.

The full message is:
> Internal Server Error ("failed to set up container networking: driver failed programming external connectivity on endpoint zealous_mayer (...): failed to bind host port for 0.0.0.0:6443:172.17.0.4:6443/tcp: address already in use")

Simplifying the check should ensure that we only see this message when necessary

Reply:
The error message still exists in the docker source code: https://github.com/moby/moby/blob/11bf52e9e331daa235d23b6cf92298481083128d/libnetwork/portallocator/portallocator.go#L36 Not sure when either is called, but I think it might make sense to keep it here